extension-ui/background.js

```diff
@@ -1,3 +1,89 @@
+let currentTab = null;
+let startTime = null;
+
+chrome.tabs.onActivated.addListener(async (activeInfo) => {
+  await updateTimeForPreviousTab();
```

Comment:
part of the code in onActivated and onUpdated is same, please use common function